pkg/providers/aws/cluster_network_provider.go

```diff
@@ -0,0 +1,87 @@
+package aws
```

Comment:
I think we need to be verbose about why this exists, for future engineers. So they understand why we have a fork in our logic.

```suggestion
// utility to manage a dedicated vpc for the resources created by the cloud resource operator.
//
// this has been added to allow the operator to work with clusters provisioned with openshift tooling >= 4.4.6,
// as they will not allow multi-az cloud resources to be created in single-az openshift clusters due to the single-az
// cluster subnets taking up all networking addresses in the cluster vpc.
//
// any openshift clusters that have used the cloud resource operator before this utility was added will be using the old approach,
// which is bundling cloud resources in with the cluster vpc. backwards compatibility for this approach must be maintained.
// 
// see [1] for more details.
//
// [1] 

package aws
```